### PR TITLE
feat: Update Methodical Mind to use the plate reader REC schema

### PR DESCRIPTION
**Pull request by joshua-benchling**
* Changed schema mapper
* Added new fields in the data system document.

### CODE REVIEW

**Review by nathan-stender — APPROVED** [review]

File: src/allotropy/parsers/methodical_mind/methodical_mind_structure.py [hunk 1]
```diff
@@ -87,12 +87,14 @@ def create(luminescence: int, location_id: str, well_plate_id: str) -> WellData:
 def create_metadata(header: Header) -> Metadata:
     return Metadata(
         file_name=header.file_name.rsplit("\\", 1)[-1],
-        unc_path=header.file_name,
+        unc_path=NOT_APPLICABLE,
```
Comment (slopez-b) [inline src/allotropy/parsers/methodical_mind/methodical_mind_structure.py:90]: Is this an expected change? We are replacing the path with `N/A`. This seems like something that should be in its own PR.

Comment (joshua-benchling) [reply src/allotropy/parsers/methodical_mind/methodical_mind_structure.py:90]: Thanks for catching up on this
